CSharp/Library/Microsoft.Bot.Builder/Dialogs/IDialogContext.cs

```diff
@@ -109,6 +149,44 @@ public static partial class Extensions
             }
 
             await botToUser.PostAsync(message, cancellationToken);
+        }
+
+
+        /// <summary>
+        /// Post a message and optional SSML to be sent to the user, using previous messages to establish a conversation context.
+        /// </summary>
+        /// <remarks>
+        /// If the locale parameter is not set, locale of the incoming message will be used for reply.
+        /// </remarks>
+        /// <param name="botToUser">Communication channel to use.</param>
+        /// <param name="text">The message text.</param>
+        /// <param name="speak">The SSML markup for text to speech.</param>
+        /// <param name="options">The optioms for the message.</param>
+        /// <param name="locale">The locale of the text.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A task that represents the post operation.</returns>
+        public static async Task SayAsync(this IBotToUser botToUser, string text, string speak = null, MessageOptions options = null, string locale = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var message = botToUser.MakeMessage();
+
+            message.Text = text;
+            message.Speak = speak;
+
+            if (!string.IsNullOrEmpty(locale))
+            {
+                message.Locale = locale;
+            }
+
+            if (options != null)

```

Comment:
Are you referring to the MessageOptions class? I’m fine either way, I added a class so it’s aligned to what Steve did. He has an interface [IMessageOptions](https://github.com/Microsoft/BotBuilder/blob/master/Node/core/lib/botbuilder.d.ts#L106-L122). We can change it to keyword named arguments if that makes more sense. LMK